### ISSUE TITLE
Envisakit/Homebridge 127 errors

### ISSUE THREAD

**Issue by BuddyBoyz** [mklips0/homebridge-envisakit#1]
Hello mklips0 and everyone!

I was very intrigued by the thought of controlling my Envisalink 4 (the newest model) connected to my Ademco 20P with the Homebridge and Envisakit scripts.  I carefully followed all directions on a new Rasbberry Pi 2, and installed everything.  I tried using both the Elgato Eve and iDevices apps.  

Homebridge starts up fine, and then listens for commands from Eve.  When I start up the Eve app, and navigate to my Envisakit accessory, the following error message repeats several times on the Pi when homebridge attempts to find the current status of the alarm:

[Envisakit] Getting current state via command: undefined status -j 
[Envisakit] Envisakit command failed (127), errors = /bin/sh: 1: undefined: not found

The Eve app says the alarm state is Stay Arm (incorrect), and when I go to try and change it to disarm, Homebridge attempts to disarm it with the correct pin code like this (pin changed to XXXX for privacy):

[Envisakit] Attempting to set alarm state to: disarm, via command: undefined disarm -p XXXX (alarm pin deleted for privacy) 
[Envisakit] Error - could not set alarm state, error = 127

Interestingly, when I go to a regular Terminal command line and try to access Envisakit, I can access the Envisakit module just fine:

pi@raspberrypi ~/.homebridge $ /home/pi/envisakit/envisakit-cli status
Connecting to 10.0.XX.XXX:4025  (IP chunk deleted for privacy)
Connected
Update: ****DISARMED****  Ready to Arm  
Ready
AC Present

---------------------------------

Here is the configuration file (config.json) for Homebridge:

{
    "bridge": {
        "name": "Homebridge-01",
        "username": "CC:22:3D:E3:CE:30",
        "port": 51826,
        "pin": "031-45-154"
    },
    
    "accessories": [
        {
            "accessory": "Envisakit",
            "name": "Envisakit",
            "cmd": "/home/pi/envisakit/envisakit-cli",
            "pin": "XXXX", (deleted for privacy)
            "periodic_update": false,
            "cache_timeout": 15,
            "args": "",
            "bypass_zone": "13",
            "show_alarm_detected": true,
            "show_chime": true,
            "show_battery_level": true
        }
    ]
}

---------------------------------

So to summarize, Homebridge starts up fine and waits for commands from a Homebridge app (i.e. Elgato Eve or iDevice).  Envisakit can operate the alarm just fine from the Pi terminal command line when I enter the Envisakit commands manually.    However, those two can't talk to each other without the errors listed above (error 127).  Just to be safe, I wiped the Raspberry's drive, reinstalled everything, and followed all installation directions carefully.  Exact same errors again.

Any ideas mklips0, or the other great folks here?

**Comment by mklips0** [comment]
Hi there. Based on your description, it seems like everything is installed correctly, but there's an error in the setup guide. In your config, try changing the line `"cmd": "/home/pi/envisakit/envisakit-cli",` to `"envisakit-path": "/home/pi/envisakit/envisakit-cli",`. Then just restart homebridge - this should solve the issue. 

My apologies about the error, it's called "cmd" in the code so it was an oversight in the example config. I'll adjust the readme shortly.

Hope this helps!

**Comment by BuddyBoyz** [comment]
Thank you mklips0 - your fix worked perfectly!  

On a side note, please also change the Fields description below the configuration sample where it says ""cmd": Absolute path to your envisakit-cli script (required)" to a matching "envisakit-path"

You also have this posted at npmjs (https://www.npmjs.com/package/homebridge-envisakit) so please make the changes to the Sample configuration and Fields spots there as well.

Thank you again.  BuddyBoyz
